File: extract.js
const fs = require('fs');
const pdf = require('pdf-parse');

try {
    console.log("Reading file...");
    let dataBuffer = fs.readFileSync('C:\\Users\\kuany\\Desktop\\сайт меню\\Мәзір Сірне KZ.pdf');
    console.log("Parsing PDF...");
    pdf(dataBuffer).then(function (data) {
        fs.writeFileSync('menu-text.txt', data.text);
        console.log("SUCCESS");
    }).catch(function (error) {
        console.error("PDF_ERROR", error);
    });
} catch (e) {
    console.error("FILE_ERROR", e);
}

File: extract.py
import traceback
import io
import sys
sys.stdout.reconfigure(encoding='utf-8')
try:
    import docx
    doc = docx.Document(r'c:\Users\kuany\Desktop\сайт меню\Sirne_Menu_full.docx')
    text = '\n'.join([para.text for para in doc.paragraphs if para.text.strip()])
    open(r'c:\Users\kuany\Desktop\сайт меню\sirne-kz\menu-text.txt', 'w', encoding='utf-8').write(text)
    print('SUCCESS')
except Exception as e:
    traceback.print_exc()
